src/github.com/travis-ci/worker/backend/bluebox.go

```diff
@@ -0,0 +1,280 @@
+package backend
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pborman/uuid"
+
+	"github.com/henrikhodne/goblueboxapi"
+	"github.com/pkg/sftp"
+	"github.com/travis-ci/worker/config"
+	"github.com/travis-ci/worker/context"
+	"github.com/travis-ci/worker/metrics"
+	"golang.org/x/crypto/ssh"
+	gocontext "golang.org/x/net/context"
+)
+
+var (
+	templateRegexp = regexp.MustCompile(`travis-([\w-]+)-\d{4}-\d{2}-\d{2}-\d{2}-\d{2}`)
+	errNoBlueBoxIP = fmt.Errorf("no IP address assigned")
+)
+
+const (
+	blueBoxHelp = `
+              CUSTOMER_ID - [REQUIRED] account customer id
+                  API_KEY - [REQUIRED] API key
+              LOCATION_ID - [REQUIRED] location where job blocks will be provisioned
+               PRODUCT_ID - [REQUIRED]
+                IPV6_ONLY - boot all blocks with only an IPv6 address (default false)
+  LANGUAGE_MAP_{LANGUAGE} - Map the key specified in the key to the image associated
```

Comment:
I was on the fence about the language mapping implementation.  In my GCE branch I went ahead and did a single structured string, but that was mostly because of laziness.  I like your approach better :smiley_cat: